src/chan_vese.jl

```diff
@@ -0,0 +1,241 @@
+
+## `μ::Float64`
+
+The argument `μ` is a weight controlling the penalty on the total length
+of the curve 𝐶;
+
+For example, if the boundaries of the image are quite smooth, a larger `μ`
+can prevent 𝐶 from being a complex curve.
+
+Default: 0.25
+
+## `λ₁::Float64`, `λ₂::Float64`
+
+The argument `λ₁` and `λ₂` affect the desired uniformity inside 𝐶 and 
+outside 𝐶, respectively. 
+
+For example, if set `λ₁` < `λ₂`, we are more possible to get result with 
+quite uniform background and varying grayscale objects in the foreground.
+
+Default: λ₁ = 1.0
+         λ₂ = 1.0
+
+## `tol::Float64`
+
+The argument `tol` controls the level set variation tolerance between 
+iteration. If the L2 norm difference between two level sets of adjacent
+iterations is below `tol`, then the solution will be assumed to be reached.
+
+Default: 1e-3
+
+## `max_iter::Int`
+
+The argument `max_iter` controls the maximum of iteration number.
+
+Default: 500
+
+## `Δt::Float64`
+
+The argument `Δt` is a multiplication factor applied at calculations 
+for each step, serves to accelerate the algorithm. Although larger `Δt`
+can speed up the algorithm, it might prevent algorithm from converging to 
+the solution.
+
+Default: 0.5
+
+## reinitial_flag::Bool
+
+The arguement `reinitial_flag` controls whether to reinitialize the
+level set in each step.
+
+Default: false
+
+# Examples
+
+```julia
+using TestImages
+using ImageSegmentation
+
+img = testimage("cameraman")
+
+cv_result = chan_vese(img, max_iter=200)
+```
+
+# References
+
+[1] An Active Contour Model without Edges, Tony Chan and Luminita Vese, 
+    Scale-Space Theories in Computer Vision, 1999, :DOI:`10.1007/3-540-48236-9_13`
+[2] Chan-Vese Segmentation, Pascal Getreuer Image Processing On Line, 2 (2012), 
+    pp. 214-224, :DOI:`10.5201/ipol.2012.g-cv`
+[3] The Chan-Vese Algorithm - Project Report, Rami Cohen, 2011 :arXiv:`1107.2782`
+"""
+function chan_vese(img::GenericGrayImage;
+                   μ::Float64=0.25,
+                   λ₁::Float64=1.0,
+                   λ₂::Float64=1.0,
+                   tol::Float64=1e-3,
+                   max_iter::Int=500,
+                   Δt::Float64=0.5,
+                   reinitial_flag::Bool=false)
+    # Signs used in the codes and comments mainly follow paper[3] in the References.
+    img = float64.(channelview(img))
+    N = ndims(img)
+    iter = 0
+    h = 1.0
+    del = tol + 1
+    img .= img .- minimum(img)
+
+    if maximum(img) != 0
+        img .= img ./ maximum(img)
+    end
+
+    # Precalculation of some constants which helps simplify some integration   
+    # area = length(img) # area = ∫H𝚽 + ∫H𝚽ⁱ
+    area = length(img) # area = ∫H𝚽 + ∫H𝚽ⁱ
+    ∫u₀ = sum(img)     # ∫u₀ = ∫u₀H𝚽 + ∫u₀H𝚽ⁱ
+
+    # Initialize the level set
+    𝚽ⁿ = initial_level_set(size(img))
+
+    # Preallocation and initializtion
+    H𝚽 = trues(size(img)...)
+    𝚽ⁿ⁺¹ = similar(𝚽ⁿ)
+
+    Δ = ntuple(d -> CartesianIndex(ntuple(i -> i == d ? 1 : 0, N)), N)
```

Comment:
If you like, you can even make this
```suggestion
    Δ = ntuple(d -> CartesianIndex(ntuple(i -> i == d, N)), N)
```
since the `Bool` will be promoted to `Int`. (Optional, of course, and you may not like it better). I've probably got places I could use that myself...